rest_framework_simplejwt/views.py

```diff
@@ -28,10 +37,65 @@ def post(self, request, *args, **kwargs):
         except TokenError as e:
             raise InvalidToken(e.args[0])
 
-        return Response(serializer.validated_data, status=status.HTTP_200_OK)
+        response = Response(serializer.validated_data, status=status.HTTP_200_OK)
+
+        if api_settings.AUTH_COOKIE:
+            csrf.get_token(self.request)
+            response = self.set_auth_cookies(response, serializer.validated_data)
+
+        return response
+
+    def set_auth_cookies(self, response, data):
+        return response
 
 
-class TokenObtainPairView(TokenViewBase):
+class TokenRefreshViewBase(TokenViewBase):
+    def extract_token_from_cookie(self, request):
+        return request
+
+    def post(self, request, *args, **kwargs):
+        if api_settings.AUTH_COOKIE:
+            request = self.extract_token_from_cookie(request)
+        return super().post(request, *args, **kwargs)
+
+
+class TokenCookieViewMixin:
+    def extract_token_from_cookie(self, request):
+        token = request.COOKIES.get('{}_refresh'.format(api_settings.AUTH_COOKIE))
+        if not token:
+            raise NotAuthenticated(detail=_('Refresh cookie not set. Try to authenticate first.'))
+        else:
+            request.data['refresh'] = token
```

Comment:
This line call exception 'This QueryDict instance is immutable'

Reply:
fixed by 078224ee606352d1b72fe498f91c61bac8d2ed33